src/test/java/bot/commands/audio/VoiceChannelTest.java

```diff
@@ -38,18 +38,18 @@ public void canJoinVoiceChannelSuccessfully(){
         when(mockGuildVoiceState.inVoiceChannel()).thenReturn(true);
         when(mockGuildVoiceState.getChannel()).thenReturn(mockVoiceChannel);
 
-
-        AudioPlayerManager mockAudioPlayerManager = mock(AudioPlayerManager.class);
         AudioManager mockAudioManager = mock(AudioManager.class);
+
         ArgumentCaptor<AudioSendHandler> audioSendHandlerArgumentCaptor = ArgumentCaptor.forClass(AudioSendHandler.class);
         doAnswer(invocation -> null).when(mockAudioManager).setSendingHandler(audioSendHandlerArgumentCaptor.capture());
 
+        ArgumentCaptor<VoiceChannel> voiceChannelArgumentCaptor = ArgumentCaptor.forClass(VoiceChannel.class);
+        doAnswer(invocation -> null).when(mockAudioManager).openAudioConnection(voiceChannelArgumentCaptor.capture());
+
         Member mockMember = mock(Member.class);
         when(mockMember.getVoiceState()).thenReturn(mockGuildVoiceState);
+
         Guild mockGuild = mock(Guild.class);
         when(mockGuild.getAudioManager()).thenReturn(mockAudioManager);
-        when(mockAudioPlayerManager.createPlayer()).thenReturn();
-
-        VoiceChannelUtils.joinVoiceChannel();
```

Comment:
Use it to capture argument values for further assertions.
Mockito verifies argument values in natural java style: by using an equals() method. This is also the recommended way of matching arguments because it makes tests clean & simple. In some situations though, it is helpful to assert on certain arguments after the actual verification. For example:
     ArgumentCaptor<Person> argument = ArgumentCaptor.forClass(Person.class);
     verify(mock).doSomething(argument.capture());
     assertEquals("John", argument.getValue().getName());
   
Example of capturing varargs:
     //capturing varargs:
     ArgumentCaptor<Person> varArgs = ArgumentCaptor.forClass(Person.class);
     verify(mock).varArgMethod(varArgs.capture());
     List expected = asList(new Person("John"), new Person("Jane"));
     assertEquals(expected, varArgs.getAllValues());
   
Warning: it is recommended to use ArgumentCaptor with verification but not with stubbing. Using ArgumentCaptor with stubbing may decrease test readability because captor is created outside of assert (aka verify or 'then') block. Also it may reduce defect localization because if stubbed method was not called then no argument is captured.
In a way ArgumentCaptor is related to custom argument matchers (see javadoc for ArgumentMatcher class). Both techniques can be used for making sure certain arguments were passed to mocks. However, ArgumentCaptor may be a better fit if: